native/packages/react-native-bpk-component-image/src/BpkImage.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Animated,
+  StyleSheet,
+  View,
+  ViewPropTypes,
+  type StyleObj,
+} from 'react-native';
+import BpkSpinner from 'react-native-bpk-component-spinner';
+
+import {
+  animationDurationBase,
+  borderRadiusSm,
+  colorGray300,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  outer: {
+    width: '100%',
+    height: '100%',
+  },
+  outerWithBorderRadius: {
+    borderRadius: borderRadiusSm,
+    overflow: 'hidden',
+  },
+  image: {
+    width: '100%',
+    height: '100%',
+    position: 'absolute',
+  },
+  loadingIndicatorView: {
+    width: '100%',
+    height: '100%',
+    backgroundColor: colorGray300,
+    justifyContent: 'center',
+    alignItems: 'center',
+    position: 'absolute',
+  },
+});
+
+export type Props = {
+  onLoad: ?() => mixed,
+  style: ?StyleObj,
+  alt: string,
+  borderRadius: boolean,
+  inView: boolean,
+  loaded: boolean,
+  source: { uri: string } | (string => mixed),
+};
+
+type State = {
+  imageOpacity: number,
+  loadingIndicatorOpacity: number,
+  showLoadingIndicator: boolean,
+};
+
+class BpkImage extends Component<Props, State> {
+  static propTypes = {
+    alt: PropTypes.string.isRequired,
+    source: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
+      .isRequired,
+    borderRadius: PropTypes.bool,
+    inView: PropTypes.bool,
+    loaded: PropTypes.bool,
+    onLoad: PropTypes.func,
+    style: ViewPropTypes.style,
+  };
+
+  static defaultProps = {
+    borderRadius: true,
+    inView: true,
+    loaded: true,
+    onLoad: null,
+    style: null,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showLoadingIndicator: !props.loaded,
+      imageOpacity: new Animated.Value(props.loaded ? 1 : 0),
+      loadingIndicatorOpacity: new Animated.Value(props.loaded ? 0 : 1),
+    };
+  }
+
+  componentWillReceiveProps = newProps => {
```

Comment:
Convention is to use `nextProps` as the arg name, plus because we have a Flow type for props, you can wrap it in that:

```... (nextProps: Props) => {```

Reply:
I don't think we should be using `componentWillReceiveProps` it's advised against in the [React documentation](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops).